common/static/common/js/discussion/views/discussion_thread_list_view.js

```diff
@@ -493,29 +511,126 @@
+                                .eq(0);
+                            this.filterEnabled = false;
+                            $curOption.removeClass('is-focused');
+                            $next.addClass('is-focused');
+                            this.selectOption($next);
+                        } else {
+                            this.selectOption($next);
+                        }
+                    }
+                    break;
+                }
+
+                default: {
+                    break;
+                }
+
+                }
+                return true;
+            };
+
+            DiscussionThreadListView.prototype.filterTopics = function() {
                 var items, query,
                     self = this;
-                query = $(event.target).val();
+                query = this.$('.forum-nav-browse-filter-input').val();
                 items = this.$('.forum-nav-browse-menu-item');
                 if (query.length === 0) {
+                    items.find('.forum-nav-browse-title.is-focused').removeClass('is-focused');
                     return items.show();
                 } else {
-                    items.hide();
-                    return items.each(function(i, item) {
-                        var path, pathText,
-                            $item = $(item);
-                        if (!$item.is(':visible')) {
-                            pathText = self.getPathText($item).toLowerCase();
-                            if (query.split(' ').every(function(term) {
-                                return pathText.search(term.toLowerCase()) !== -1;
-                            })) {
-                                path = $item.parents('.forum-nav-browse-menu-item').andSelf();
-                                return path.add($item.find('.forum-nav-browse-menu-item')).show();
+                    if (this.filterEnabled) {
+                        items.hide();
+                        return items.each(function(i, item) {
+                            var path, pathText,
+                                $item = $(item);
+                            if (!$item.is(':visible')) {
+                                pathText = self.getPathText($item).toLowerCase();
+                                if (query.split(' ').every(function(term) {
+                                    return pathText.search(term.toLowerCase()) !== -1;
+                                })) {
+                                    path = $item.parents('.forum-nav-browse-menu-item').andSelf();
+                                    return path.add($item.find('.forum-nav-browse-menu-item')).show();
+                                }
                             }
-                        }
-                    });
+                            return items;
+                        });
+                    }
                 }
+                return true;
```

Comment:
Eslint is making you return a value on this line because you return `items.show()` if `query.length===0` (line 613) and `items` if `query.length !== 0 && this.filterEnabled` (line 629) but you don't return anything if `query.length !== 0 && !this.filterEnabled`

The consistent-return rule makes it so either all or none of the branches of a conditional inside a function result inside a return. Technically adding `return true` on the last line makes that linter warning go away, but I think the real fix is to figure out what this function should return if `query.length !== 0 && !this.filterEnabled`

Likewise for `DiscussionThreadListView.prototype.keyboardBinding` and its return on line 603 above.